x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/template/TemplateUtils.java

```diff
@@ -59,9 +59,10 @@ public static void loadTemplateIntoMap(String resource, Map<String, IndexTemplat
     public static String loadTemplate(String resource, String version, String versionProperty) {
         try {
             BytesReference source = load(resource);
-            validate(source);
+            final String filteredJson = filter(source, version, versionProperty);
+            validate(new BytesArray(filteredJson));
```

Comment:
Yes, bare placeholders (e.g. `"version": ${my.variable}`) didn't work with the validation before the variable replacement, and version must be a bare integer, not a string. Previously, all variables were strings (e.g. `"field": "${my.variable}"`, note the quotes around the variable).

Besides, it's more important that the JSON is valid *after* we do the variable replacement, rather than before, isn't it?

Reply:
It is a misleading name - I would have taken the opportunity to change it, but Dimitris is going to do so in https://github.com/elastic/elasticsearch/pull/51765, as part of modifying `IndexTemplateRegistry` (and associated classes) to allow other variable substitutions in addition to the version.